### PR TITLE
[IOS-422]: Retrieve prekeys, regester user with chat service if 401 returned

### PR DESCRIPTION
**Pull request by yuliaveres**
Also:
remove not needed key from UserDefaults wrapper;
do not store network requests password in database, create new one each login;
refresh prekeys if there are less then desirable amount left (defines by constant of `20`, same as used in bots code).

### CODE REVIEW

File: Toshi/Supporting Files/AppDelegate.m [hunk 1]
```diff
@@ -380,32 +382,58 @@ - (void)configureForCurrentSession
...
-                                                         [CrashlyticsLogger log:@"Successfully registered user with chat service after forced trigger" attributes:nil];
-                                                     } else {
-                                                         [CrashlyticsLogger log:@"Failed to register user with chat service after forced trigger" attributes:nil];
-                                                     }
-                                                 }];
-                                             }
-                                         }];
-    }
+    NSURL *url = [NSURL URLWithString:[NSString stringWithFormat:@"%@",
+                                       textSecureKeysAPI]];
```
Comment (designatednerd) [inline Toshi/Supporting Files/AppDelegate.m]: if this is a string you should just be able to pass it in directly without the `stringWithFormat`

File: Toshi/Supporting Files/AppDelegate.m [hunk 2]
```diff
@@ -380,32 +382,57 @@ - (void)configureForCurrentSession
...
+                                         if (error.code == 401) {
+                                             [ChatAPIClient.shared registerUserWithCompletion:^(BOOL success) {
+                                                 if (success) {
+                                                     [CrashlyticsLogger log:@"Successfully registered user with chat service after forced trigger" attributes:nil];
+                                                 } else {
+                                                     [CrashlyticsLogger log:@"Failed to register user with chat service after forced trigger" attributes:nil];
+                                                 }
+                                             }];
+                                         }
+                                     }];
```
Comment (designatednerd) [inline Toshi/Supporting Files/AppDelegate.m]: Is it worth pulling this out into some kind of separate service for testing (or at least getting it out of the app delegate)?

Comment (yuliaveres) [reply Toshi/Supporting Files/AppDelegate.m]: i wont do it now. so won;t close pr until i do

File: Toshi/Supporting Files/PreKeyHandler.m [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+#import "PrekeysRequest.h"
+#import <SignalServiceKit/TSNetworkManager.h>
+#import <SignalServiceKit/TSPreKeyManager.h>
+#import "Toshi-Swift.h"
+
+NSUInteger const PREKEY_MINIMUM_COUNT = 20;
+
+@implementation PreKeyHandler
+
++ (void)tryRetrievingPrekeys
```
Comment (designatednerd) [inline Toshi/Supporting Files/PreKeyHandler.m]: is there any way to test any of this?

Comment (yuliaveres) [reply Toshi/Supporting Files/PreKeyHandler.m]: there is always way to test this. we might be back to it later when there is more time

Comment (designatednerd) [reply Toshi/Supporting Files/PreKeyHandler.m]: but what if by adding tests now we prevent more mistakes with this and save ourselves even more time? 🤔 🤔 🤔

**Pull request merged**